modules/travel_pay/spec/controllers/claims_controller_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TravelPay::ClaimsController, type: :request do
+  let(:user) { build(:user) }
+
+  before do
+    veis_response = double
+    allow(veis_response).to receive(:body).and_return( {'access_token' => 'sample_token'} )
+    allow_any_instance_of(TravelPay::Client).to receive(:request_veis_token).and_return(veis_response)
+    btsss_ping_response = double
+    allow(btsss_ping_response).to receive(:status).and_return('online')
+    allow_any_instance_of(TravelPay::Client).to receive(:ping).and_return(btsss_ping_response)
+    sign_in(user)
+    Flipper.disable :travel_pay_power_switch
+  end
+
+  describe '#index' do
+    context 'the feature switch is enabled' do
+      before do
+        Flipper.enable :travel_pay_power_switch
+      end
+
+      it 'requests a token and sends a ping to BTSSS' do
+        get '/travel_pay/claims'
+        expect(response.body).to include('Received ping from upstream server with status online.')
```

Comment:
I like to expect inclusion of the smallest parts of strings. For example, instead of the whole string: "Received ping from upstream server with status online.", what about

```
expect(response.body).to include('ping')
```

The thought being we'll probably always include the word "ping" here, but the other parts could change.

What are your thoughts on this?

Reply:
Or, if you want to take any string-inclusionary guesswork out of the equation, you could try to expect that the `TravelPay::Client#ping` mock method was called.

```
expect(TravelPay::Client).to receive(:ping)
```

This requires some constructor dependency injection, but possibly worth it. Thoughts?

Reply:
I'll include both of these suggestions - I think it's important to test both that the method was called and the response.